main.py

```diff
@@ -9,79 +9,105 @@
 from colorama import Fore
 from datetime import datetime
 
-# Press the green button in the gutter to run the script.
-if __name__ == '__main__':
-    #parser
-    my_parser=argparse.ArgumentParser()
-    my_parser.add_argument('function',
-                           action='store',
-                           choices=['sort','search','filter','count'],
-                           help='set the function to use on zip file')
-    my_parser.add_argument('arg_of_function',
-                           action='store',
-                           default='NULL')
-    my_parser.add_argument('-additional_data_first',
-                           action='store',
-                           help='data by wich you filter')
-    my_parser.add_argument('-additional_data_second',
-                           action='store',
-                           help='data by wich you filter')
-    args=my_parser.parse_args()
-    function=args.function
 
-    #reading from the zip file
-    with ZipFile('myFile0.zip') as zf:
-        with zf.open('myFile0.csv', 'r') as data:
-            reader = csv.reader(TextIOWrapper(data, 'utf-8'))
-            header = next(reader)
-            rows = []
-            row = []
-            for row in reader:
-                rows.append(row)
-    # functions to do
-    if function == 'sort':
-        col = args.arg_of_function
-        result=sorted(rows,key=lambda row: row[int(col)-1])
-        for i in range(0, 10):
-            print('\n'+ str(result[i]))
+class Functionality:
+    def __init__(self, function, arg_of_function, additional_data_first, additional_data_second, rows):
+        self.function = function
+        self.arg_of_function = arg_of_function
+        self.additional_data_first = additional_data_first
+        self.additional_data_second = additional_data_second
+        self.rows = rows
+
+    def result_of_function(self):
+        if self.function == 'sort':
+            col = self.arg_of_function
+            result = sorted(self.rows, key=lambda row: row[int(col) - 1])
+            for i in range(0, 10):
```

Comment:
You can also write this for statement like this `print(result[:10], sep='\n')` by using list slicing. It's a pretty powerful feature.

Reply:
No need to change it tho. I just saw a simplier way to do it